debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java

```diff
@@ -500,6 +503,144 @@ protected SourceRecords consumeRecordsByTopic(int numRecords, boolean assertReco
+     * @return the actual number of Dml records that were consumed
+     * @throws InterruptedException if the thread was interrupted while waiting for a record to be returned
+     */
+    protected int consumeDmlRecordsByTopic(int numberDmlRecords, Consumer<SourceRecord> recordConsumer) throws InterruptedException {
+        return consumeDmlRecordsByTopic(numberDmlRecords, 3, recordConsumer, true);
+    }
+
+    /**
+     * Try to consume the specified number of records from the connector, calling the given function for each, and return the
+     * actual number of Dml records that were consumed.
+     *
+     * For slower connectors it is possible to receive no records from the connector at most 3 times in a row
+     * until the waiting is terminated.  Additionally, while this method will consume and append transaction metadata
+     * topic records to the consumer, the returned value only considers Dml records.
+     *
+     * @param numberOfRecords the number of Dml records that should be consumed
+     * @param breakAfterNulls the number of allowed run when no records are consumed
+     * @param recordConsumer the function that should be called for each consumed record
+     * @param assertRecords true if records serialization should be verified
+     * @return the actual number of Dml records that were consumed
+     * @throws InterruptedException if the thread was interrupted while waiting for a record to be returned
+     */
+    protected int consumeDmlRecordsByTopic(int numberOfRecords, int breakAfterNulls, Consumer<SourceRecord> recordConsumer, boolean assertRecords)
+            throws InterruptedException {
+        int recordsConsumed = 0;
+        int nullReturn = 0;
+        Set<String> endTransactions = new LinkedHashSet<>();
+        while (recordsConsumed < numberOfRecords) {
+            SourceRecord record = consumedLines.poll(pollTimeoutInMs, TimeUnit.MILLISECONDS);
+            if (record != null) {
+                nullReturn = 0;
+                if (isTransactionRecord(record)) {
+                    final Struct value = (Struct) record.value();
+                    final String status = value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_STATUS_KEY);
+                    if (status.equals(TransactionStatus.BEGIN.name())) {
+                        endTransactions.add(value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_ID_KEY));
+                    }
+                    else {
+                        endTransactions.remove(value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_ID_KEY));
+                    }
+                }
+                else {
+                    ++recordsConsumed;
+                }
+                if (recordConsumer != null) {
+                    recordConsumer.accept(record);
+                }
+                if (Testing.Debug.isEnabled()) {
+                    Testing.debug("Consumed record " + recordsConsumed + " / " + numberOfRecords + " ("
```

Comment:
Could the log messages here and down contain also information about `endTransactions`?